caffe2/python/memonger_test.py

```diff
@@ -483,10 +483,10 @@ def test_rnn(self):
                 "cell_init_{}".format(i)
             )
             init_blobs.extend([hidden_init, cell_init])
-        model.param_init_net.ConstantFill([], ["input"], shape=[T, 4, 10])
+        input_blob = model.param_init_net.ConstantFill([], ["input"], shape=[T, 4, 10])
```

Comment:
why do you have to change it? Did it stop working with string blob names?

Reply:
Yes, it stopped working because of this change: https://github.com/caffe2/caffe2/pull/2025/files#diff-def726fb1d234a67aeaf10e07aa06da7R995

What happens is that we scope `self.cells[0].prepare_input` with `self.name` (not with layer id) and that causes all the inputs and outputs of `FC` in `prepare_input` to be scoped unless they are given as strings instead of BlobReference. If we want to keep `input_blob` out of the scope of the LSTM, then we have to use a BlobReference, otherwise the test will crash unless we scope https://github.com/caffe2/caffe2/blob/master/caffe2/python/memonger_test.py#L486 as well.

Reply:
Ah, I see. So a trick I did before when applying scopes is the following. If any input is a string, we rectify it before entering a scope. Here is an example. https://github.com/caffe2/caffe2/blob/master/caffe2/python/rnn_cell.py#L182

In your case you just need to take of one single input_blob :)